### ISSUE TITLE
CGROUP does not get deleted on failure.

### ISSUE THREAD

**Issue by pcdummy** [lxc/lxd#2764]
This is not a common bug but a IMHO a bug:

When LXD fails to create a container `Failed to set up id mapping.` the cgroup gets never deleted:

    error: Error calling 'lxd forkstart psql1 /var/lib/lxd/containers /var/log/lxd/psql1/lxc.conf': err='exit status 1'
    lxc 20161224084734.754 ERROR lxc_cgfsng - cgroups/cgfsng.c:create_path_for_hierarchy:1310 - Path "/sys/fs/cgroup/systemd//lxc/psql1" already existed.
    lxc 20161224084734.754 ERROR lxc_cgfsng - cgroups/cgfsng.c:cgfsng_create:1367 - No such file or directory - Failed to create /sys/fs/cgroup/systemd//lxc/psql1: No such file or directory

# Required information

 * Distribution: Ubuntu
 * Distribution version: Xenial
 * The output of "lxc info" or if that fails:
   * Kernel version: 4.4.0-57-generic
   * LXC version: 2.6.2
   * LXD version: 2.6.2
   * Storage backend in use: BTRFS

# Steps to reproduce

 1. Mess up id mapping (mess up /etc/sub?id
 2. Run a existing container with now bad uid mapping
 3. Fix uid mapping
 4. Run it again.

**Comment by pcdummy** [comment]
I'm only able to reproduce this with debian jessie containers.

**Comment by stgraber** [comment]
Can you post an actual "lxc info" output?

**Comment by pcdummy** [comment]
Ofc, sorry. In the meanwhile i updated LXD/LXC to 2.7:

```
root@srv01:~# lxc info
apiextensions:
- storage_zfs_remove_snapshots
- container_host_shutdown_timeout
- container_syscall_filtering
- auth_pki
- container_last_used_at
- etag
- patch
- usb_devices
- https_allowed_credentials
- image_compression_algorithm
- directory_manipulation
- container_cpu_time
- storage_zfs_use_refquota
- storage_lvm_mount_options
- network
- profile_usedby
- container_push
- container_exec_recording
- certificate_update
- container_exec_signal_handling
- gpu_devices
- container_image_properties
- migration_progress
- id_map
- network_firewall_filtering
- network_routes
apistatus: stable
apiversion: "1.0"
auth: trusted
environment:
  addresses:
  - 10.167.160.14:8443
  - '[fd57:c87d:f1ee:ee00:f::14]:8443'
  - '[hidden]:8443'
  - 10.0.3.1:8443
  - 192.168.122.1:8443
  architectures:
  - x86_64
  - i686
  certificate: |
    -----BEGIN CERTIFICATE-----
    -----END CERTIFICATE-----
  certificatefingerprint: 1ec2bcdfa5288bbc7d2a857a42278bc8392e2e16b2d481b3f104131ceb33d5c7
  driver: lxc
  driverversion: 2.0.6
  kernel: Linux
  kernelarchitecture: x86_64
  kernelversion: 4.4.0-57-generic
  server: lxd
  serverpid: 1410
  serverversion: "2.7"
  storage: btrfs
  storageversion: "4.4"
config:
  core.https_address: '[::]:8443'
  core.trust_password: true
public: false
```

**Comment by stgraber** [comment]
And you can still reproduce it with the above LXD correct?

@brauner any idea what's going on? Looks like LXC doesn't clean after itself in that case and also fails to reuse somehow? (it's liblxc 2.0.6)

**Comment by brauner** [comment]
@pcdummy, can you please append the complete log for the container? Also, if this is reproducible, please run `LXD` in debug mode, reproduce and append the logs for this container (since they will have `DEBUG` level logging enabled then).

**Comment by brauner** [comment]
If at all, this is likely caused by having incorrect permissions due to the messed-up uid-mappings @pcdummy was referencing. So `cgfsng_destroy()` fails in it's call to `userns_exec_1()` on container exec. When the next container starts with a fixed uid-mapping it obviously cannot remove or modify the already existing cgroups because they do not have the same uid-mapping as it "usually" has. Note that this shouldn't be a problem since `cgfsng()` will try `1000` times to create a usable cgroup for the container. So I'm weary to call this an actual bug.

**Comment by stgraber** [comment]
@pcdummy is this still a problem? Can you attach the information @brauner asked for?

**Comment by gaetanquentin** [comment]
i think this issue is still here, in lxd 2.12 (ubuntu 16.10):

config:
  core.https_address: '[::]'
  core.trust_password: true
api_extensions:
- storage_zfs_remove_snapshots
- container_host_shutdown_timeout
- container_syscall_filtering
- auth_pki
- container_last_used_at
- etag
- patch
- usb_devices
- https_allowed_credentials
- image_compression_algorithm
- directory_manipulation
- container_cpu_time
- storage_zfs_use_refquota
- storage_lvm_mount_options
- network
- profile_usedby
- container_push
- container_exec_recording
- certificate_update
- container_exec_signal_handling
- gpu_devices
- container_image_properties
- migration_progress
- id_map
- network_firewall_filtering
- network_routes
- storage
- file_delete
- file_append
- network_dhcp_expiry
- storage_lvm_vg_rename
- storage_lvm_thinpool_rename
- network_vlan
- image_create_aliases
- container_stateless_copy
api_status: stable
api_version: "1.0"
auth: trusted
public: false
environment:
  addresses:
  - 172.16.99.16:8443
  - 172.17.10.1:8443
  - 172.17.22.1:8443
  - 172.17.11.1:8443
  - 172.17.12.1:8443
  architectures:
  - i686
  certificate: |
  certificate_fingerprint: 7e36584daa9cab0e74d86b2e5b359d5c31e3a253db429576b450e0aaba768985
  driver: lxc
  driver_version: 2.0.7
  kernel: Linux
  kernel_architecture: i686
  kernel_version: 4.8.0-42-generic
  server: lxd
  server_pid: 28521
  server_version: "2.12"
  storage: btrfs
  storage_version: 4.7.3



same steps to reproduce than @pcdummy 

output in debug mode: (second time):


DBUG[03-23|23:34:49] handling                                 ip=@ method=GET url=/1.0
DBUG[03-23|23:34:49] Initializing a BTRFS driver. 
DBUG[03-23|23:34:49] 
	{
		"type": "sync",
		"status": "Success",
		"status_code": 200,
		"operation": "",
		"error_code": 0,
		"error": "",
		"metadata": {
			"config": {
				"core.https_address": "[::]",
				"core.trust_password": true
			},
			"api_extensions": [
				"storage_zfs_remove_snapshots",
				"container_host_shutdown_timeout",
				"container_syscall_filtering",
				"auth_pki",
				"container_last_used_at",
				"etag",
				"patch",
				"usb_devices",
				"https_allowed_credentials",
				"image_compression_algorithm",
				"directory_manipulation",
				"container_cpu_time",
				"storage_zfs_use_refquota",
				"storage_lvm_mount_options",
				"network",
				"profile_usedby",
				"container_push",
				"container_exec_recording",
				"certificate_update",
				"container_exec_signal_handling",
				"gpu_devices",
				"container_image_properties",
				"migration_progress",
				"id_map",
				"network_firewall_filtering",
				"network_routes",
				"storage",
				"file_delete",
				"file_append",
				"network_dhcp_expiry",
				"storage_lvm_vg_rename",
				"storage_lvm_thinpool_rename",
				"network_vlan",
				"image_create_aliases",
				"container_stateless_copy"
			],
			"api_status": "stable",
			"api_version": "1.0",
			"auth": "trusted",
			"public": false,
			"environment": {
				"addresses": [
					"172.16.99.16:8443",
					"172.17.10.1:8443",
					"172.17.22.1:8443",
					"172.17.11.1:8443",
					"172.17.12.1:8443"
				],
				"architectures": [
					"i686"
				],
				"certificate": "-----BEGIN CERTIFICATE-----\nMIIFTzCCAzegAwIBAgIQZTLxMoF1Dso8662hj4TP5jANBgkqhkiG9w0BAQsFADAx\nMRwwGgYDVQQKExNsaW51eGNvbnRhaW5lcnMub3JnMREwDwYDVQQDDAhyb290QG5h\nczAeFw0xNzAzMTMyMDMyMzVaFw0yNzAzMTEyMDMyMzVaMDExHDAaBgNVBAoTE2xp\nbnV4Y29udGFpbmVycy5vcmcxETAPBgNVBAMMCHJvb3RAbmFzMIICIjANBgkqhkiG\n9w0BAQEFAAOCAg8AMIICCgKCAgEAsJp9ZLAE9PHYlCGlAT6xEc5PAwK12ozMPeKI\nL4DBNa+vmjxYzLuH7ppOVKRL+PTaXgjKzGZ9PC2IQ9/4EaQ72qc/Y2RqrJ520nJB\nGxJWDFY9OeGBY0dFe7Cmjqe3yP6RdsLb0OvnBFeZUqR+Uy3tuAsSXwAU6n3uqaRl\ng8p2AqV/6D/+mBg3quwECbWT+4KP9u9BstVL2qiw5EmvR8c5tV0KbE7L5NMnUGYi\n55nt1OHhu6jEieKNoOm1tewNtMzBcJ6pFIIHrq9yAS6nMEQQ99NNXK9oCWC0mi/9\n4J7ZBCQjZmntUgkGAaZML+GDXwaeEwebM66YaBnrSba7YuE6aI5smSTmXNK+dSU9\n1i9lpUpSUlyYQEmWumM5kOmJzQNWTtOweW+55k62Vb53zUdrnQHknaDHSzt2QUQf\nzhlPg7G649j6UxRbt3l3TIxjRX4xXA7d4arXWIRx686hmZOPNxvnBLrDLwz9YFUD\nkpCph45iP65qFrSpF3rN0JUEh0EtH+az0LrZDaCKiAC9n7OATt6rLBY4vurc7MIq\nYzDt2ANXaOMb3txsTlpL/9wu4Kpj3WsSQ+ye422e3DXN3LcybZ/DQMQOnNlh4la3\nYvrAZ2eEFBe9BnjFyJSlcjG6nKc4vSOvF9pHayrlKm2QVADP5oOT2vcFL/Q3xSvp\nf6JVCUkCAwEAAaNjMGEwDgYDVR0PAQH/BAQDAgWgMBMGA1UdJQQMMAoGCCsGAQUF\nBwMBMAwGA1UdEwEB/wQCMAAwLAYDVR0RBCUwI4IDbmFzhwSsEGMQhwSsEQoBhwSs\nERYBhwSsEQsBhwSsEQwBMA0GCSqGSIb3DQEBCwUAA4ICAQBGmIBAlMAX+5pVg3Ru\nG1YkCriKyneECltTnM6NHOaDPXTIZC78+WVSspDveatudVdzFIBkNMhxsJ+M6kh3\nUhu8TJx7O8ifcC0vbwsFsYXaElt4CuUbKw2EJ23uq6ttcPVoPhSUkgMJEFki0R1N\nPq7KAo6xTNmpqa3cr3hN+X9qdKiOVixwag9HvXoVYB5tT6kn/YcEyLal93Oh131S\nPyztgb/mQhoH6I1dgs+QEHZvtj4fXMqfoXF9gwb6wjL5axKPjttathNsezeb+cuC\noECoOXWN6yCCvA4ZMknPU6y8T3kuBIPiUw3HJsQ67nwXhK7Ie638KDvxC/h2qF/h\nAiTQEGilcBfBe0yL4t7v6jQtTwtDBl2l03zWR1tKilL4lWj6s9B6m6TXvv3CU7yz\nkPH/sdNzNodyqj/abbAjaCWavWhNM3nq/3sbvS9TDxPJ4uDGszcOE4WU4eU9k2Bk\nJCp+djO9bcm4A/K3QrNb5+DuNgbIsygKeXV0dIR2NB4bKziE4bvdqYyWj/HQz7qB\nHAdPbOyoIpy5dW7ijXZ80SARHrb8/G1uJ0+tKp2Zv2YrIrgj5kMWQCFUNgaO6sAA\nb7Btp/+PXoZ868FhpmuC5E06++9s4Bq/XZI5C9qloNe1jC+gZ8IbMNE5+4dU6L2E\n4MOXDXi0vPre0N3D35WNb1+E+A==\n-----END CERTIFICATE-----\n",
				"certificate_fingerprint": "",
				"driver": "lxc",
				"driver_version": "2.0.7",
				"kernel": "Linux",
				"kernel_architecture": "i686",
				"kernel_version": "4.8.0-42-generic",
				"server": "lxd",
				"server_pid": 31175,
				"server_version": "2.12",
				"storage": "btrfs",
				"storage_version": "4.7.3"
			}
		}
	} 
DBUG[03-23|23:34:49] handling                                 ip=@ method=GET url=/1.0/containers/mycontainer
DBUG[03-23|23:34:49] 
	{
		"type": "sync",
		"status": "Success",
		"status_code": 200,
		"operation": "",
		"error_code": 0,
		"error": "",
		"metadata": {
			"architecture": "i686",
			"config": {
				"image.architecture": "i386",
				"image.description": "Ubuntu yakkety i386 (20170312_03:49)",
				"image.os": "Ubuntu",
				"image.release": "yakkety",
				"image.serial": "20170312_03:49",
				"raw.idmap": "both 1003 1003",
				"volatile.base_image": "4d574fd7ecf8d7f758cb212d92e8fef12876a8c0f03e5048cabbf179482919c7",
				"volatile.eth0.hwaddr": "00:16:3e:73:67:8f",
				"volatile.idmap.base": "0",
				"volatile.idmap.next": "[{\"Isuid\":true,\"Isgid\":false,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":1003},{\"Isuid\":true,\"Isgid\":true,\"Hostid\":1003,\"Nsid\":1003,\"Maprange\":1},{\"Isuid\":true,\"Isgid\":false,\"Hostid\":101004,\"Nsid\":1004,\"Maprange\":64532},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":1003},{\"Isuid\":true,\"Isgid\":true,\"Hostid\":1003,\"Nsid\":1003,\"Maprange\":1},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":101004,\"Nsid\":1004,\"Maprange\":64532}]",
				"volatile.last_state.idmap": "[{\"Isuid\":true,\"Isgid\":false,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":1003},{\"Isuid\":true,\"Isgid\":true,\"Hostid\":1003,\"Nsid\":1003,\"Maprange\":1},{\"Isuid\":true,\"Isgid\":false,\"Hostid\":101004,\"Nsid\":1004,\"Maprange\":64532},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":1003},{\"Isuid\":true,\"Isgid\":true,\"Hostid\":1003,\"Nsid\":1003,\"Maprange\":1},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":101004,\"Nsid\":1004,\"Maprange\":64532}]",
				"volatile.last_state.power": "RUNNING"
			},
			"devices": {
				"download_disk_mycontainer": {
					"path": "/data/Downloads",
					"source": "/mnt/UUID_5770585f-2ac3-4681-a3ca-93d6287165f5/@medias/@mycontainer",
					"type": "disk"
				}
			},
			"ephemeral": false,
			"profiles": [
				"default",
				"anonymous"
			],
			"created_at": "2017-03-14T13:54:30Z",
			"expanded_config": {
				"image.architecture": "i386",
				"image.description": "Ubuntu yakkety i386 (20170312_03:49)",
				"image.os": "Ubuntu",
				"image.release": "yakkety",
				"image.serial": "20170312_03:49",
				"raw.idmap": "both 1003 1003",
				"volatile.base_image": "4d574fd7ecf8d7f758cb212d92e8fef12876a8c0f03e5048cabbf179482919c7",
				"volatile.eth0.hwaddr": "00:16:3e:73:67:8f",
				"volatile.idmap.base": "0",
				"volatile.idmap.next": "[{\"Isuid\":true,\"Isgid\":false,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":1003},{\"Isuid\":true,\"Isgid\":true,\"Hostid\":1003,\"Nsid\":1003,\"Maprange\":1},{\"Isuid\":true,\"Isgid\":false,\"Hostid\":101004,\"Nsid\":1004,\"Maprange\":64532},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":1003},{\"Isuid\":true,\"Isgid\":true,\"Hostid\":1003,\"Nsid\":1003,\"Maprange\":1},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":101004,\"Nsid\":1004,\"Maprange\":64532}]",
				"volatile.last_state.idmap": "[{\"Isuid\":true,\"Isgid\":false,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":1003},{\"Isuid\":true,\"Isgid\":true,\"Hostid\":1003,\"Nsid\":1003,\"Maprange\":1},{\"Isuid\":true,\"Isgid\":false,\"Hostid\":101004,\"Nsid\":1004,\"Maprange\":64532},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":1003},{\"Isuid\":true,\"Isgid\":true,\"Hostid\":1003,\"Nsid\":1003,\"Maprange\":1},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":101004,\"Nsid\":1004,\"Maprange\":64532}]",
				"volatile.last_state.power": "RUNNING"
			},
			"expanded_devices": {
				"download_disk_mycontainer": {
					"path": "/data/Downloads",
					"source": "/mnt/UUID_5770585f-2ac3-4681-a3ca-93d6287165f5/@medias/@mycontainer",
					"type": "disk"
				},
				"eth0": {
					"name": "eth0",
					"nictype": "bridged",
					"parent": "unsec0",
					"type": "nic"
				},
				"root": {
					"path": "/",
					"pool": "default",
					"type": "disk"
				}
			},
			"name": "mycontainer",
			"stateful": false,
			"status": "Stopped",
			"status_code": 102,
			"last_used_at": "2017-03-23T22:34:08.870916229Z"
		}
	} 
DBUG[03-23|23:34:49] handling                                 ip=@ method=PUT url=/1.0/containers/mycontainer/state
DBUG[03-23|23:34:49] 
	{
		"action": "start",
		"force": false,
		"stateful": false,
		"timeout": 0
	} 
DBUG[03-23|23:34:49] New task operation: a51a6cc9-1208-4ff3-a770-c4415af9aceb 
DBUG[03-23|23:34:49] Started task operation: a51a6cc9-1208-4ff3-a770-c4415af9aceb 
DBUG[03-23|23:34:49] 
	{
		"type": "async",
		"status": "Operation created",
		"status_code": 100,
		"operation": "/1.0/operations/a51a6cc9-1208-4ff3-a770-c4415af9aceb",
		"error_code": 0,
		"error": "",
		"metadata": {
			"id": "a51a6cc9-1208-4ff3-a770-c4415af9aceb",
			"class": "task",
			"created_at": "2017-03-23T23:34:49.857567601+01:00",
			"updated_at": "2017-03-23T23:34:49.857567601+01:00",
			"status": "Running",
			"status_code": 103,
			"resources": {
				"containers": [
					"/1.0/containers/mycontainer"
				]
			},
			"metadata": null,
			"may_cancel": false,
			"err": ""
		}
	} 
DBUG[03-23|23:34:49] handling                                 ip=@ method=GET url=/1.0/operations/a51a6cc9-1208-4ff3-a770-c4415af9aceb/wait
DBUG[03-23|23:34:49] Initializing a BTRFS driver. 
DBUG[03-23|23:34:52] Mounting BTRFS storage volume for container "mycontainer" on storage pool "default". 
DBUG[03-23|23:34:52] Mounting BTRFS storage pool "default". 
DBUG[03-23|23:34:52] Mounted BTRFS storage volume for container "mycontainer" on storage pool "default". 
DBUG[03-23|23:34:52] Mounting BTRFS storage volume for container "mycontainer" on storage pool "default". 
DBUG[03-23|23:34:52] Mounting BTRFS storage pool "default". 
DBUG[03-23|23:34:52] Mounted BTRFS storage volume for container "mycontainer" on storage pool "default". 
INFO[03-23|23:34:52] Starting container                       action=start created=2017-03-14T13:54:30+0000 ephemeral=false name=mycontainer stateful=false used=2017-03-23T22:34:08+0000
DBUG[03-23|23:34:52] handling                                 ip=@ method=GET url=/1.0
DBUG[03-23|23:34:53] Initializing a BTRFS driver. 
DBUG[03-23|23:34:53] 
	{
		"type": "sync",
		"status": "Success",
		"status_code": 200,
		"operation": "",
		"error_code": 0,
		"error": "",
		"metadata": {
			"config": {
				"core.https_address": "[::]",
				"core.trust_password": true
			},
			"api_extensions": [
				"storage_zfs_remove_snapshots",
				"container_host_shutdown_timeout",
				"container_syscall_filtering",
				"auth_pki",
				"container_last_used_at",
				"etag",
				"patch",
				"usb_devices",
				"https_allowed_credentials",
				"image_compression_algorithm",
				"directory_manipulation",
				"container_cpu_time",
				"storage_zfs_use_refquota",
				"storage_lvm_mount_options",
				"network",
				"profile_usedby",
				"container_push",
				"container_exec_recording",
				"certificate_update",
				"container_exec_signal_handling",
				"gpu_devices",
				"container_image_properties",
				"migration_progress",
				"id_map",
				"network_firewall_filtering",
				"network_routes",
				"storage",
				"file_delete",
				"file_append",
				"network_dhcp_expiry",
				"storage_lvm_vg_rename",
				"storage_lvm_thinpool_rename",
				"network_vlan",
				"image_create_aliases",
				"container_stateless_copy"
			],
			"api_status": "stable",
			"api_version": "1.0",
			"auth": "trusted",
			"public": false,
			"environment": {
				"addresses": [
					"172.16.99.16:8443",
					"172.17.10.1:8443",
					"172.17.22.1:8443",
					"172.17.11.1:8443",
					"172.17.12.1:8443"
				],
				"architectures": [
					"i686"
				],
				"certificate": "-----BEGIN CERTIFICATE-----\nMIIFTzCCAzegAwIBAgIQZTLxMoF1Dso8662hj4TP5jANBgkqhkiG9w0BAQsFADAx\nMRwwGgYDVQQKExNsaW51eGNvbnRhaW5lcnMub3JnMREwDwYDVQQDDAhyb290QG5h\nczAeFw0xNzAzMTMyMDMyMzVaFw0yNzAzMTEyMDMyMzVaMDExHDAaBgNVBAoTE2xp\nbnV4Y29udGFpbmVycy5vcmcxETAPBgNVBAMMCHJvb3RAbmFzMIICIjANBgkqhkiG\n9w0BAQEFAAOCAg8AMIICCgKCAgEAsJp9ZLAE9PHYlCGlAT6xEc5PAwK12ozMPeKI\nL4DBNa+vmjxYzLuH7ppOVKRL+PTaXgjKzGZ9PC2IQ9/4EaQ72qc/Y2RqrJ520nJB\nGxJWDFY9OeGBY0dFe7Cmjqe3yP6RdsLb0OvnBFeZUqR+Uy3tuAsSXwAU6n3uqaRl\ng8p2AqV/6D/+mBg3quwECbWT+4KP9u9BstVL2qiw5EmvR8c5tV0KbE7L5NMnUGYi\n55nt1OHhu6jEieKNoOm1tewNtMzBcJ6pFIIHrq9yAS6nMEQQ99NNXK9oCWC0mi/9\n4J7ZBCQjZmntUgkGAaZML+GDXwaeEwebM66YaBnrSba7YuE6aI5smSTmXNK+dSU9\n1i9lpUpSUlyYQEmWumM5kOmJzQNWTtOweW+55k62Vb53zUdrnQHknaDHSzt2QUQf\nzhlPg7G649j6UxRbt3l3TIxjRX4xXA7d4arXWIRx686hmZOPNxvnBLrDLwz9YFUD\nkpCph45iP65qFrSpF3rN0JUEh0EtH+az0LrZDaCKiAC9n7OATt6rLBY4vurc7MIq\nYzDt2ANXaOMb3txsTlpL/9wu4Kpj3WsSQ+ye422e3DXN3LcybZ/DQMQOnNlh4la3\nYvrAZ2eEFBe9BnjFyJSlcjG6nKc4vSOvF9pHayrlKm2QVADP5oOT2vcFL/Q3xSvp\nf6JVCUkCAwEAAaNjMGEwDgYDVR0PAQH/BAQDAgWgMBMGA1UdJQQMMAoGCCsGAQUF\nBwMBMAwGA1UdEwEB/wQCMAAwLAYDVR0RBCUwI4IDbmFzhwSsEGMQhwSsEQoBhwSs\nERYBhwSsEQsBhwSsEQwBMA0GCSqGSIb3DQEBCwUAA4ICAQBGmIBAlMAX+5pVg3Ru\nG1YkCriKyneECltTnM6NHOaDPXTIZC78+WVSspDveatudVdzFIBkNMhxsJ+M6kh3\nUhu8TJx7O8ifcC0vbwsFsYXaElt4CuUbKw2EJ23uq6ttcPVoPhSUkgMJEFki0R1N\nPq7KAo6xTNmpqa3cr3hN+X9qdKiOVixwag9HvXoVYB5tT6kn/YcEyLal93Oh131S\nPyztgb/mQhoH6I1dgs+QEHZvtj4fXMqfoXF9gwb6wjL5axKPjttathNsezeb+cuC\noECoOXWN6yCCvA4ZMknPU6y8T3kuBIPiUw3HJsQ67nwXhK7Ie638KDvxC/h2qF/h\nAiTQEGilcBfBe0yL4t7v6jQtTwtDBl2l03zWR1tKilL4lWj6s9B6m6TXvv3CU7yz\nkPH/sdNzNodyqj/abbAjaCWavWhNM3nq/3sbvS9TDxPJ4uDGszcOE4WU4eU9k2Bk\nJCp+djO9bcm4A/K3QrNb5+DuNgbIsygKeXV0dIR2NB4bKziE4bvdqYyWj/HQz7qB\nHAdPbOyoIpy5dW7ijXZ80SARHrb8/G1uJ0+tKp2Zv2YrIrgj5kMWQCFUNgaO6sAA\nb7Btp/+PXoZ868FhpmuC5E06++9s4Bq/XZI5C9qloNe1jC+gZ8IbMNE5+4dU6L2E\n4MOXDXi0vPre0N3D35WNb1+E+A==\n-----END CERTIFICATE-----\n",
				"certificate_fingerprint": "",
				"driver": "lxc",
				"driver_version": "2.0.7",
				"kernel": "Linux",
				"kernel_architecture": "i686",
				"kernel_version": "4.8.0-42-generic",
				"server": "lxd",
				"server_pid": 31175,
				"server_version": "2.12",
				"storage": "btrfs",
				"storage_version": "4.7.3"
			}
		}
	} 
DBUG[03-23|23:34:53] handling                                 ip=@ method=GET url=/internal/containers/3/onstart
DBUG[03-23|23:34:53] Initializing a BTRFS driver. 
DBUG[03-23|23:34:53] Mounting BTRFS storage volume for container "mycontainer" on storage pool "default". 
DBUG[03-23|23:34:53] Mounting BTRFS storage pool "default". 
DBUG[03-23|23:34:53] Mounted BTRFS storage volume for container "mycontainer" on storage pool "default". 
DBUG[03-23|23:34:53] Scheduler: container mycontainer started: re-balancing 
DBUG[03-23|23:34:53] 
	{
		"type": "sync",
		"status": "Success",
		"status_code": 200,
		"operation": "",
		"error_code": 0,
		"error": "",
		"metadata": {}
	} 
DBUG[03-23|23:34:54] handling                                 ip=@ method=GET url=/1.0
DBUG[03-23|23:34:54] Initializing a BTRFS driver. 
DBUG[03-23|23:34:54] 
	{
		"type": "sync",
		"status": "Success",
		"status_code": 200,
		"operation": "",
		"error_code": 0,
		"error": "",
		"metadata": {
			"config": {
				"core.https_address": "[::]",
				"core.trust_password": true
			},
			"api_extensions": [
				"storage_zfs_remove_snapshots",
				"container_host_shutdown_timeout",
				"container_syscall_filtering",
				"auth_pki",
				"container_last_used_at",
				"etag",
				"patch",
				"usb_devices",
				"https_allowed_credentials",
				"image_compression_algorithm",
				"directory_manipulation",
				"container_cpu_time",
				"storage_zfs_use_refquota",
				"storage_lvm_mount_options",
				"network",
				"profile_usedby",
				"container_push",
				"container_exec_recording",
				"certificate_update",
				"container_exec_signal_handling",
				"gpu_devices",
				"container_image_properties",
				"migration_progress",
				"id_map",
				"network_firewall_filtering",
				"network_routes",
				"storage",
				"file_delete",
				"file_append",
				"network_dhcp_expiry",
				"storage_lvm_vg_rename",
				"storage_lvm_thinpool_rename",
				"network_vlan",
				"image_create_aliases",
				"container_stateless_copy"
			],
			"api_status": "stable",
			"api_version": "1.0",
			"auth": "trusted",
			"public": false,
			"environment": {
				"addresses": [
					"172.16.99.16:8443",
					"172.17.10.1:8443",
					"172.17.22.1:8443",
					"172.17.11.1:8443",
					"172.17.12.1:8443"
				],
				"architectures": [
					"i686"
				],
				"certificate": "-----BEGIN CERTIFICATE-----\nMIIFTzCCAzegAwIBAgIQZTLxMoF1Dso8662hj4TP5jANBgkqhkiG9w0BAQsFADAx\nMRwwGgYDVQQKExNsaW51eGNvbnRhaW5lcnMub3JnMREwDwYDVQQDDAhyb290QG5h\nczAeFw0xNzAzMTMyMDMyMzVaFw0yNzAzMTEyMDMyMzVaMDExHDAaBgNVBAoTE2xp\nbnV4Y29udGFpbmVycy5vcmcxETAPBgNVBAMMCHJvb3RAbmFzMIICIjANBgkqhkiG\n9w0BAQEFAAOCAg8AMIICCgKCAgEAsJp9ZLAE9PHYlCGlAT6xEc5PAwK12ozMPeKI\nL4DBNa+vmjxYzLuH7ppOVKRL+PTaXgjKzGZ9PC2IQ9/4EaQ72qc/Y2RqrJ520nJB\nGxJWDFY9OeGBY0dFe7Cmjqe3yP6RdsLb0OvnBFeZUqR+Uy3tuAsSXwAU6n3uqaRl\ng8p2AqV/6D/+mBg3quwECbWT+4KP9u9BstVL2qiw5EmvR8c5tV0KbE7L5NMnUGYi\n55nt1OHhu6jEieKNoOm1tewNtMzBcJ6pFIIHrq9yAS6nMEQQ99NNXK9oCWC0mi/9\n4J7ZBCQjZmntUgkGAaZML+GDXwaeEwebM66YaBnrSba7YuE6aI5smSTmXNK+dSU9\n1i9lpUpSUlyYQEmWumM5kOmJzQNWTtOweW+55k62Vb53zUdrnQHknaDHSzt2QUQf\nzhlPg7G649j6UxRbt3l3TIxjRX4xXA7d4arXWIRx686hmZOPNxvnBLrDLwz9YFUD\nkpCph45iP65qFrSpF3rN0JUEh0EtH+az0LrZDaCKiAC9n7OATt6rLBY4vurc7MIq\nYzDt2ANXaOMb3txsTlpL/9wu4Kpj3WsSQ+ye422e3DXN3LcybZ/DQMQOnNlh4la3\nYvrAZ2eEFBe9BnjFyJSlcjG6nKc4vSOvF9pHayrlKm2QVADP5oOT2vcFL/Q3xSvp\nf6JVCUkCAwEAAaNjMGEwDgYDVR0PAQH/BAQDAgWgMBMGA1UdJQQMMAoGCCsGAQUF\nBwMBMAwGA1UdEwEB/wQCMAAwLAYDVR0RBCUwI4IDbmFzhwSsEGMQhwSsEQoBhwSs\nERYBhwSsEQsBhwSsEQwBMA0GCSqGSIb3DQEBCwUAA4ICAQBGmIBAlMAX+5pVg3Ru\nG1YkCriKyneECltTnM6NHOaDPXTIZC78+WVSspDveatudVdzFIBkNMhxsJ+M6kh3\nUhu8TJx7O8ifcC0vbwsFsYXaElt4CuUbKw2EJ23uq6ttcPVoPhSUkgMJEFki0R1N\nPq7KAo6xTNmpqa3cr3hN+X9qdKiOVixwag9HvXoVYB5tT6kn/YcEyLal93Oh131S\nPyztgb/mQhoH6I1dgs+QEHZvtj4fXMqfoXF9gwb6wjL5axKPjttathNsezeb+cuC\noECoOXWN6yCCvA4ZMknPU6y8T3kuBIPiUw3HJsQ67nwXhK7Ie638KDvxC/h2qF/h\nAiTQEGilcBfBe0yL4t7v6jQtTwtDBl2l03zWR1tKilL4lWj6s9B6m6TXvv3CU7yz\nkPH/sdNzNodyqj/abbAjaCWavWhNM3nq/3sbvS9TDxPJ4uDGszcOE4WU4eU9k2Bk\nJCp+djO9bcm4A/K3QrNb5+DuNgbIsygKeXV0dIR2NB4bKziE4bvdqYyWj/HQz7qB\nHAdPbOyoIpy5dW7ijXZ80SARHrb8/G1uJ0+tKp2Zv2YrIrgj5kMWQCFUNgaO6sAA\nb7Btp/+PXoZ868FhpmuC5E06++9s4Bq/XZI5C9qloNe1jC+gZ8IbMNE5+4dU6L2E\n4MOXDXi0vPre0N3D35WNb1+E+A==\n-----END CERTIFICATE-----\n",
				"certificate_fingerprint": "",
				"driver": "lxc",
				"driver_version": "2.0.7",
				"kernel": "Linux",
				"kernel_architecture": "i686",
				"kernel_version": "4.8.0-42-generic",
				"server": "lxd",
				"server_pid": 31175,
				"server_version": "2.12",
				"storage": "btrfs",
				"storage_version": "4.7.3"
			}
		}
	} 
DBUG[03-23|23:34:54] handling                                 ip=@ method=GET url="/internal/containers/3/onstop?target=stop"
EROR[03-23|23:34:54] stop hook failed                         container=mycontainer err="Container is already running a start operation"
DBUG[03-23|23:34:54] 
	{
		"error": "Container is already running a start operation",
		"error_code": 500,
		"type": "error"
	} 
EROR[03-23|23:34:59] Failed starting container                action=start created=2017-03-14T13:54:30+0000 ephemeral=false name=mycontainer stateful=false used=2017-03-23T22:34:08+0000
DBUG[03-23|23:34:59] Failure for task operation: a51a6cc9-1208-4ff3-a770-c4415af9aceb: Failed to run: /usr/bin/lxd forkstart mycontainer /var/lib/lxd/containers /var/log/lxd/mycontainer/lxc.conf:  
DBUG[03-23|23:34:59] 
	{
		"type": "sync",
		"status": "Success",
		"status_code": 200,
		"operation": "",
		"error_code": 0,
		"error": "",
		"metadata": {
			"id": "a51a6cc9-1208-4ff3-a770-c4415af9aceb",
			"class": "task",
			"created_at": "2017-03-23T23:34:49.857567601+01:00",
			"updated_at": "2017-03-23T23:34:49.857567601+01:00",
			"status": "Failure",
			"status_code": 400,
			"resources": {
				"containers": [
					"/1.0/containers/mycontainer"
				]
			},
			"metadata": null,
			"may_cancel": false,
			"err": "Failed to run: /usr/bin/lxd forkstart mycontainer /var/lib/lxd/containers /var/log/lxd/mycontainer/lxc.conf: "
		}
	} 



lxd.log:
lvl=info msg="Expiring log files" t=2017-03-23T11:39:58+0100
lvl=info msg="Done expiring log files" t=2017-03-23T11:39:58+0100
lvl=info msg="Pruning expired images" t=2017-03-23T11:39:59+0100
lvl=info msg="Done pruning expired images" t=2017-03-23T11:39:59+0100
lvl=info msg="Updating images" t=2017-03-23T11:40:02+0100
lvl=info msg="Done updating images" t=2017-03-23T11:40:02+0100
lvl=info msg="Updating images" t=2017-03-23T17:40:02+0100
lvl=info msg="Done updating images" t=2017-03-23T17:40:02+0100
action=shutdown created=2017-03-14T13:54:30+0000 ephemeral=false lvl=info msg="Shutting down container" name=mycontainer t=2017-03-23T23:09:24+0100 timeout=-1s used=2017-03-23T05:26:45+0000
action=stop created=2017-03-14T13:54:30+0000 ephemeral=false lvl=info msg="Container initiated stop" name=mycontainer stateful=false t=2017-03-23T23:10:02+0100 used=2017-03-23T05:26:45+0000
action=shutdown created=2017-03-14T13:54:30+0000 ephemeral=false lvl=info msg="Shut down container" name=mycontainer t=2017-03-23T23:10:02+0100 timeout=-1s used=2017-03-23T05:26:45+0000
action=shutdown created=2017-03-14T04:47:59+0000 ephemeral=false lvl=info msg="Shutting down container" name=showfinders t=2017-03-23T23:10:10+0100 timeout=-1s used=2017-03-18T13:44:37+0000
action=stop created=2017-03-14T04:47:59+0000 ephemeral=false lvl=info msg="Container initiated stop" name=showfinders stateful=false t=2017-03-23T23:11:44+0100 used=2017-03-18T13:44:37+0000
action=shutdown created=2017-03-14T04:47:59+0000 ephemeral=false lvl=info msg="Shut down container" name=showfinders t=2017-03-23T23:11:44+0100 timeout=-1s used=2017-03-18T13:44:37+0000
lvl=info msg="Received 'terminated signal', exiting." t=2017-03-23T23:15:25+0100
lvl=info msg="Stopping REST API handler:" t=2017-03-23T23:15:25+0100
lvl=info msg=" - closing socket" socket=[::]:8443 t=2017-03-23T23:15:25+0100
lvl=info msg=" - skipping socket-activated socket" socket=/var/lib/lxd/unix.socket t=2017-03-23T23:15:25+0100
lvl=info msg="Stopping /dev/lxd handler" t=2017-03-23T23:15:25+0100
lvl=info msg="Stopped /dev/lxd handler" t=2017-03-23T23:15:25+0100
lvl=info msg="Unmounting temporary filesystems" t=2017-03-23T23:15:26+0100
lvl=info msg="Done unmounting temporary filesystems" t=2017-03-23T23:15:26+0100
lvl=info msg="Closing the database" t=2017-03-23T23:15:26+0100
lvl=info msg="Saving simplestreams cache" t=2017-03-23T23:15:26+0100
lvl=info msg="Saved simplestreams cache" t=2017-03-23T23:15:26+0100
lvl=info msg="LXD 2.12 is starting in normal mode" path=/var/lib/lxd t=2017-03-23T23:16:24+0100
lvl=warn msg="CGroup memory swap accounting is disabled, swap limits will be ignored." t=2017-03-23T23:16:24+0100
lvl=info msg="Kernel uid/gid map:" t=2017-03-23T23:16:24+0100
lvl=info msg=" - u 0 0 4294967295" t=2017-03-23T23:16:24+0100
lvl=info msg=" - g 0 0 4294967295" t=2017-03-23T23:16:24+0100
lvl=info msg="Configured LXD uid/gid map:" t=2017-03-23T23:16:24+0100
lvl=info msg=" - u 0 100000 65536" t=2017-03-23T23:16:24+0100
lvl=info msg=" - g 0 100000 65536" t=2017-03-23T23:16:24+0100
lvl=info msg="Expiring log files" t=2017-03-23T23:16:24+0100
lvl=info msg="Done expiring log files" t=2017-03-23T23:16:24+0100
lvl=info msg="Starting /dev/lxd handler" t=2017-03-23T23:16:24+0100
lvl=info msg="LXD is socket activated" t=2017-03-23T23:16:24+0100
lvl=info msg="REST API daemon:" t=2017-03-23T23:16:24+0100
lvl=info msg=" - binding Unix socket" socket=/var/lib/lxd/unix.socket t=2017-03-23T23:16:24+0100
lvl=info msg=" - binding TCP socket" socket=[::]:8443 t=2017-03-23T23:16:24+0100
lvl=info msg="Pruning expired images" t=2017-03-23T23:16:24+0100
lvl=info msg="Done pruning expired images" t=2017-03-23T23:16:24+0100
lvl=info msg="Updating images" t=2017-03-23T23:16:24+0100
lvl=info msg="Done updating images" t=2017-03-23T23:16:24+0100
action=start created=2017-03-14T13:54:30+0000 ephemeral=false lvl=info msg="Starting container" name=mycontainer stateful=false t=2017-03-23T23:20:46+0100 used=2017-03-23T05:26:45+0000
action=start created=2017-03-14T13:54:30+0000 ephemeral=false lvl=info msg="Started container" name=mycontainer stateful=false t=2017-03-23T23:20:47+0100 used=2017-03-23T05:26:45+0000
action=shutdown created=2017-03-14T13:54:30+0000 ephemeral=false lvl=info msg="Shutting down container" name=mycontainer t=2017-03-23T23:21:27+0100 timeout=-1s used=2017-03-23T22:20:45+0000
action=stop created=2017-03-14T13:54:30+0000 ephemeral=false lvl=info msg="Container initiated stop" name=mycontainer stateful=false t=2017-03-23T23:22:03+0100 used=2017-03-23T22:20:45+0000
action=shutdown created=2017-03-14T13:54:30+0000 ephemeral=false lvl=info msg="Shut down container" name=mycontainer t=2017-03-23T23:22:03+0100 timeout=-1s used=2017-03-23T22:20:45+0000
action=start created=2017-03-14T13:54:30+0000 ephemeral=false lvl=info msg="Starting container" name=mycontainer stateful=false t=2017-03-23T23:26:31+0100 used=2017-03-23T22:20:45+0000
container=mycontainer err="Container is already running a start operation" lvl=eror msg="stop hook failed" t=2017-03-23T23:26:32+0100
action=start created=2017-03-14T13:54:30+0000 ephemeral=false lvl=eror msg="Failed starting container" name=mycontainer stateful=false t=2017-03-23T23:26:37+0100 used=2017-03-23T22:20:45+0000
action=start created=2017-03-14T13:54:30+0000 ephemeral=false lvl=info msg="Starting container" name=mycontainer stateful=false t=2017-03-23T23:28:14+0100 used=2017-03-23T22:26:31+0000
container=mycontainer err="Container is already running a start operation" lvl=eror msg="stop hook failed" t=2017-03-23T23:28:15+0100
action=start created=2017-03-14T13:54:30+0000 ephemeral=false lvl=eror msg="Failed starting container" name=mycontainer stateful=false t=2017-03-23T23:28:20+0100 used=2017-03-23T22:26:31+0000
lvl=info msg="Received 'terminated signal', exiting." t=2017-03-23T23:33:07+0100
lvl=info msg="Stopping REST API handler:" t=2017-03-23T23:33:07+0100
lvl=info msg=" - closing socket" socket=[::]:8443 t=2017-03-23T23:33:07+0100
lvl=info msg=" - skipping socket-activated socket" socket=/var/lib/lxd/unix.socket t=2017-03-23T23:33:07+0100
lvl=info msg="Stopping /dev/lxd handler" t=2017-03-23T23:33:07+0100
lvl=info msg="Stopped /dev/lxd handler" t=2017-03-23T23:33:07+0100
lvl=info msg="Unmounting temporary filesystems" t=2017-03-23T23:33:07+0100
lvl=info msg="Done unmounting temporary filesystems" t=2017-03-23T23:33:07+0100
lvl=info msg="Closing the database" t=2017-03-23T23:33:07+0100
lvl=info msg="Saving simplestreams cache" t=2017-03-23T23:33:07+0100
lvl=info msg="Saved simplestreams cache" t=2017-03-23T23:33:07+0100



lxc container log:
            lxc 20170323223453.495 INFO     lxc_cgroup - cgroups/cgroup.c:cgroup_init:68 - cgroup driver cgroupfs-ng initing for mycontainer
            lxc 20170323223453.496 ERROR    lxc_cgfsng - cgroups/cgfsng.c:create_path_for_hierarchy:1306 - Path "/sys/fs/cgroup/systemd//lxc/mycontainer" already existed.
            lxc 20170323223453.496 ERROR    lxc_cgfsng - cgroups/cgfsng.c:cgfsng_create:1363 - No such file or directory - Failed to create /sys/fs/cgroup/systemd//lxc/mycontainer: No such file or directory
            lxc 20170323223453.496 ERROR    lxc_cgfsng - cgroups/cgfsng.c:create_path_for_hierarchy:1306 - Path "/sys/fs/cgroup/systemd//lxc/mycontainer-1" already existed.
            lxc 20170323223453.496 ERROR    lxc_cgfsng - cgroups/cgfsng.c:cgfsng_create:1363 - No such file or directory - Failed to create /sys/fs/cgroup/systemd//lxc/mycontainer-1: No such file or directory
            lxc 20170323223453.496 ERROR    lxc_cgfsng - cgroups/cgfsng.c:create_path_for_hierarchy:1306 - Path "/sys/fs/cgroup/systemd//lxc/mycontainer-2" already existed.
            lxc 20170323223453.496 ERROR    lxc_cgfsng - cgroups/cgfsng.c:cgfsng_create:1363 - No such file or directory - Failed to create /sys/fs/cgroup/systemd//lxc/mycontainer-2: No such file or directory
            lxc 20170323223453.498 DEBUG    lxc_cgfsng - cgroups/cgfsng.c:filter_and_set_cpus:474 - No isolated cpus detected.
            lxc 20170323223453.498 DEBUG    lxc_cgfsng - cgroups/cgfsng.c:handle_cpuset_hierarchy:644 - "cgroup.clone_children" was already set to "1".

**Comment by brauner** [comment]
If you ran `LXD` in debug mode can you please paste the whole container's log?

**Comment by gaetanquentin** [comment]
lxc 20170323223452.854 DEBUG    lxc_monitor - monitor.c:lxc_monitord_spawn:309 - Going to wait for pid 31608.
            lxc 20170323223452.855 DEBUG    lxc_monitor - monitor.c:lxc_monitord_spawn:328 - Trying to sync with child process.
            lxc 20170323223452.856 INFO     lxc_start - start.c:lxc_check_inherited:235 - Closed inherited fd: 3.
            lxc 20170323223452.856 INFO     lxc_start - start.c:lxc_check_inherited:235 - Closed inherited fd: 7.
            lxc 20170323223452.857 DEBUG    lxc_monitor - monitor.c:lxc_monitord_spawn:366 - Using pipe file descriptor 5 for monitord.
            lxc 20170323223452.898 DEBUG    lxc_monitor - monitor.c:lxc_monitord_spawn:343 - Sucessfully synced with child process.
            lxc 20170323223452.900 DEBUG    lxc_monitor - monitor.c:lxc_monitord_spawn:312 - Finished waiting on pid 31608.
            lxc 20170323223452.901 INFO     lxc_container - lxccontainer.c:do_lxcapi_start:804 - Attempting to set proc title to [lxc monitor] /var/lib/lxd/containers mycontainer
            lxc 20170323223452.903 INFO     lxc_start - start.c:lxc_check_inherited:235 - Closed inherited fd: 7.
            lxc 20170323223452.903 INFO     lxc_lsm - lsm/lsm.c:lsm_init:48 - LSM security driver AppArmor
            lxc 20170323223452.904 INFO     lxc_seccomp - seccomp.c:parse_config_v2:402 - processing: .reject_force_umount  # comment this to allow umount -f;  not recommended.
            lxc 20170323223452.904 INFO     lxc_seccomp - seccomp.c:parse_config_v2:553 - Adding native rule for reject_force_umount action 0.
            lxc 20170323223452.904 INFO     lxc_seccomp - seccomp.c:do_resolve_add_rule:251 - Setting Seccomp rule to reject force umounts.
            lxc 20170323223452.904 INFO     lxc_seccomp - seccomp.c:parse_config_v2:402 - processing: .[all].
            lxc 20170323223452.904 INFO     lxc_seccomp - seccomp.c:parse_config_v2:402 - processing: .kexec_load errno 38.
            lxc 20170323223452.904 INFO     lxc_seccomp - seccomp.c:parse_config_v2:553 - Adding native rule for kexec_load action 327718.
            lxc 20170323223452.905 INFO     lxc_seccomp - seccomp.c:parse_config_v2:402 - processing: .open_by_handle_at errno 38.
            lxc 20170323223452.905 INFO     lxc_seccomp - seccomp.c:parse_config_v2:553 - Adding native rule for open_by_handle_at action 327718.
            lxc 20170323223452.905 INFO     lxc_seccomp - seccomp.c:parse_config_v2:402 - processing: .init_module errno 38.
            lxc 20170323223452.905 INFO     lxc_seccomp - seccomp.c:parse_config_v2:553 - Adding native rule for init_module action 327718.
            lxc 20170323223452.905 INFO     lxc_seccomp - seccomp.c:parse_config_v2:402 - processing: .finit_module errno 38.
            lxc 20170323223452.905 INFO     lxc_seccomp - seccomp.c:parse_config_v2:553 - Adding native rule for finit_module action 327718.
            lxc 20170323223452.905 INFO     lxc_seccomp - seccomp.c:parse_config_v2:402 - processing: .delete_module errno 38.
            lxc 20170323223452.905 INFO     lxc_seccomp - seccomp.c:parse_config_v2:553 - Adding native rule for delete_module action 327718.
            lxc 20170323223452.905 INFO     lxc_conf - conf.c:run_script_argv:424 - Executing script "/usr/bin/lxd callhook /var/lib/lxd 3 start" for container "mycontainer", config section "lxc".
            lxc 20170323223452.905 DEBUG    lxc_monitor - monitor.c:lxc_monitord_spawn:309 - Going to wait for pid 31612.
            lxc 20170323223452.907 DEBUG    lxc_monitor - monitor.c:lxc_monitord_spawn:328 - Trying to sync with child process.
            lxc 20170323223452.908 INFO     lxc_start - start.c:lxc_check_inherited:235 - Closed inherited fd: 3.
            lxc 20170323223452.909 INFO     lxc_start - start.c:lxc_check_inherited:235 - Closed inherited fd: 7.
            lxc 20170323223452.909 DEBUG    lxc_monitor - monitor.c:lxc_monitord_spawn:366 - Using pipe file descriptor 5 for monitord.
            lxc 20170323223452.935 DEBUG    lxc_monitor - monitor.c:lxc_monitord_spawn:343 - Sucessfully synced with child process.
            lxc 20170323223452.938 DEBUG    lxc_monitor - monitor.c:lxc_monitord_spawn:312 - Finished waiting on pid 31612.
            lxc 20170323223452.938 INFO     lxc_monitor - monitor.c:lxc_monitor_sock_name:185 - Using monitor socket name "lxc/d78a9d7e97b4b375//var/lib/lxd/containers".
            lxc 20170323223453.462 DEBUG    lxc_start - start.c:setup_signal_fd:273 - Set SIGCHLD handler with file descriptor: 5.
            lxc 20170323223453.464 DEBUG    console - console.c:lxc_console_peer_default:468 - no console peer
            lxc 20170323223453.464 INFO     lxc_start - start.c:lxc_init:475 - Container "mycontainer" is initialized.
            lxc 20170323223453.466 DEBUG    lxc_start - start.c:__lxc_start:1317 - Not dropping CAP_SYS_BOOT or watching utmp.
            lxc 20170323223453.478 INFO     lxc_conf - conf.c:instantiate_veth:2647 - Retrieved mtu 1500 from unsec0
            lxc 20170323223453.492 INFO     lxc_conf - conf.c:instantiate_veth:2672 - Attached 'veth8YJW9F': to the bridge 'unsec0': 
            lxc 20170323223453.494 DEBUG    lxc_conf - conf.c:instantiate_veth:2689 - instantiated veth 'veth8YJW9F/vethK8CRC7', index is '53'
            lxc 20170323223453.495 INFO     lxc_cgroup - cgroups/cgroup.c:cgroup_init:68 - cgroup driver cgroupfs-ng initing for mycontainer
            lxc 20170323223453.496 ERROR    lxc_cgfsng - cgroups/cgfsng.c:create_path_for_hierarchy:1306 - Path "/sys/fs/cgroup/systemd//lxc/mycontainer" already existed.
            lxc 20170323223453.496 ERROR    lxc_cgfsng - cgroups/cgfsng.c:cgfsng_create:1363 - No such file or directory - Failed to create /sys/fs/cgroup/systemd//lxc/mycontainer: No such file or directory
            lxc 20170323223453.496 ERROR    lxc_cgfsng - cgroups/cgfsng.c:create_path_for_hierarchy:1306 - Path "/sys/fs/cgroup/systemd//lxc/mycontainer-1" already existed.
            lxc 20170323223453.496 ERROR    lxc_cgfsng - cgroups/cgfsng.c:cgfsng_create:1363 - No such file or directory - Failed to create /sys/fs/cgroup/systemd//lxc/mycontainer-1: No such file or directory
            lxc 20170323223453.496 ERROR    lxc_cgfsng - cgroups/cgfsng.c:create_path_for_hierarchy:1306 - Path "/sys/fs/cgroup/systemd//lxc/mycontainer-2" already existed.
            lxc 20170323223453.496 ERROR    lxc_cgfsng - cgroups/cgfsng.c:cgfsng_create:1363 - No such file or directory - Failed to create /sys/fs/cgroup/systemd//lxc/mycontainer-2: No such file or directory
            lxc 20170323223453.498 DEBUG    lxc_cgfsng - cgroups/cgfsng.c:filter_and_set_cpus:474 - No isolated cpus detected.
            lxc 20170323223453.498 DEBUG    lxc_cgfsng - cgroups/cgfsng.c:handle_cpuset_hierarchy:644 - "cgroup.clone_children" was already set to "1".
            lxc 20170323223453.509 INFO     lxc_start - start.c:lxc_spawn:1154 - Cloned CLONE_NEWUSER.
            lxc 20170323223453.509 INFO     lxc_start - start.c:lxc_spawn:1154 - Cloned CLONE_NEWNS.
            lxc 20170323223453.510 INFO     lxc_start - start.c:lxc_spawn:1154 - Cloned CLONE_NEWPID.
            lxc 20170323223453.510 INFO     lxc_start - start.c:lxc_spawn:1154 - Cloned CLONE_NEWUTS.
            lxc 20170323223453.510 INFO     lxc_start - start.c:lxc_spawn:1154 - Cloned CLONE_NEWIPC.
            lxc 20170323223453.514 INFO     lxc_confile - confile.c:config_idmap:1531 - read uid map: type u nsid 0 hostid 100000 range 1003
            lxc 20170323223453.514 INFO     lxc_confile - confile.c:config_idmap:1531 - read uid map: type u nsid 1003 hostid 1003 range 1
            lxc 20170323223453.515 INFO     lxc_confile - confile.c:config_idmap:1531 - read uid map: type g nsid 1003 hostid 1003 range 1
            lxc 20170323223453.515 INFO     lxc_confile - confile.c:config_idmap:1531 - read uid map: type u nsid 1004 hostid 101004 range 64532
            lxc 20170323223453.515 INFO     lxc_confile - confile.c:config_idmap:1531 - read uid map: type g nsid 0 hostid 100000 range 1003
            lxc 20170323223453.515 INFO     lxc_confile - confile.c:config_idmap:1531 - read uid map: type g nsid 1004 hostid 101004 range 64532
            lxc 20170323223453.574 ERROR    lxc_start - start.c:lxc_spawn:1171 - Failed to set up id mapping.
            lxc 20170323223453.620 INFO     lxc_conf - conf.c:lxc_delete_network:3015 - Removed interface "eth0" with index 53.
            lxc 20170323223453.636 WARN     lxc_conf - conf.c:lxc_delete_network:3038 - Failed to remove "veth8YJW9F" from host: Invalid argument.
            lxc 20170323223453.638 ERROR    lxc_start - start.c:__lxc_start:1346 - Failed to spawn container "mycontainer".
            lxc 20170323223453.672 INFO     lxc_conf - conf.c:run_script_argv:424 - Executing script "/usr/share/lxcfs/lxc.reboot.hook" for container "mycontainer", config section "lxc".
            lxc 20170323223454.225 INFO     lxc_conf - conf.c:run_script_argv:424 - Executing script "/usr/bin/lxd callhook /var/lib/lxd 3 stop" for container "mycontainer", config section "lxc".
            lxc 20170323223454.318 ERROR    lxc_conf - conf.c:run_buffer:405 - Script exited with status 1.
            lxc 20170323223454.318 ERROR    lxc_start - start.c:lxc_fini:546 - Failed to run lxc.hook.post-stop for container "mycontainer".
            lxc 20170323223454.318 WARN     lxc_commands - commands.c:lxc_cmd_rsp_recv:172 - Command get_cgroup failed to receive response: Connection reset by peer.
            lxc 20170323223454.319 WARN     lxc_commands - commands.c:lxc_cmd_rsp_recv:172 - Command get_cgroup failed to receive response: Connection reset by peer.
            lxc 20170323223454.332 ERROR    lxc_conf - conf.c:userns_exec_1:4453 - Error setting up child mappings
            lxc 20170323223454.332 ERROR    lxc_cgfsng - cgroups/cgfsng.c:recursive_destroy:1270 - Error destroying /sys/fs/cgroup/systemd//lxc/mycontainer-3
            lxc 20170323223454.347 ERROR    lxc_conf - conf.c:userns_exec_1:4453 - Error setting up child mappings
            lxc 20170323223454.347 ERROR    lxc_cgfsng - cgroups/cgfsng.c:recursive_destroy:1270 - Error destroying /sys/fs/cgroup/cpu//lxc/mycontainer-3
            lxc 20170323223454.358 ERROR    lxc_conf - conf.c:userns_exec_1:4453 - Error setting up child mappings
            lxc 20170323223454.358 ERROR    lxc_cgfsng - cgroups/cgfsng.c:recursive_destroy:1270 - Error destroying /sys/fs/cgroup/hugetlb//lxc/mycontainer-3
            lxc 20170323223454.369 ERROR    lxc_conf - conf.c:userns_exec_1:4453 - Error setting up child mappings
            lxc 20170323223454.369 ERROR    lxc_cgfsng - cgroups/cgfsng.c:recursive_destroy:1270 - Error destroying /sys/fs/cgroup/cpuset//lxc/mycontainer-3
            lxc 20170323223454.380 ERROR    lxc_conf - conf.c:userns_exec_1:4453 - Error setting up child mappings
            lxc 20170323223454.380 ERROR    lxc_cgfsng - cgroups/cgfsng.c:recursive_destroy:1270 - Error destroying /sys/fs/cgroup/perf_event//lxc/mycontainer-3
            lxc 20170323223454.391 ERROR    lxc_conf - conf.c:userns_exec_1:4453 - Error setting up child mappings
            lxc 20170323223454.391 ERROR    lxc_cgfsng - cgroups/cgfsng.c:recursive_destroy:1270 - Error destroying /sys/fs/cgroup/devices//lxc/mycontainer-3
            lxc 20170323223454.402 ERROR    lxc_conf - conf.c:userns_exec_1:4453 - Error setting up child mappings
            lxc 20170323223454.402 ERROR    lxc_cgfsng - cgroups/cgfsng.c:recursive_destroy:1270 - Error destroying /sys/fs/cgroup/blkio//lxc/mycontainer-3
            lxc 20170323223454.412 ERROR    lxc_conf - conf.c:userns_exec_1:4453 - Error setting up child mappings
            lxc 20170323223454.412 ERROR    lxc_cgfsng - cgroups/cgfsng.c:recursive_destroy:1270 - Error destroying /sys/fs/cgroup/net_cls//lxc/mycontainer-3
            lxc 20170323223454.423 ERROR    lxc_conf - conf.c:userns_exec_1:4453 - Error setting up child mappings
            lxc 20170323223454.423 ERROR    lxc_cgfsng - cgroups/cgfsng.c:recursive_destroy:1270 - Error destroying /sys/fs/cgroup/freezer//lxc/mycontainer-3
            lxc 20170323223454.434 ERROR    lxc_conf - conf.c:userns_exec_1:4453 - Error setting up child mappings
            lxc 20170323223454.434 ERROR    lxc_cgfsng - cgroups/cgfsng.c:recursive_destroy:1270 - Error destroying /sys/fs/cgroup/memory//lxc/mycontainer-3
            lxc 20170323223454.445 ERROR    lxc_conf - conf.c:userns_exec_1:4453 - Error setting up child mappings
            lxc 20170323223454.445 ERROR    lxc_cgfsng - cgroups/cgfsng.c:recursive_destroy:1270 - Error destroying /sys/fs/cgroup/pids//lxc/mycontainer-3
            lxc 20170323224319.394 INFO     lxc_confile - confile.c:config_idmap:1531 - read uid map: type u nsid 0 hostid 100000 range 1003
            lxc 20170323224319.395 INFO     lxc_confile - confile.c:config_idmap:1531 - read uid map: type u nsid 1003 hostid 1003 range 1
            lxc 20170323224319.395 INFO     lxc_confile - confile.c:config_idmap:1531 - read uid map: type g nsid 1003 hostid 1003 range 1
            lxc 20170323224319.395 INFO     lxc_confile - confile.c:config_idmap:1531 - read uid map: type u nsid 1004 hostid 101004 range 64532
            lxc 20170323224319.395 INFO     lxc_confile - confile.c:config_idmap:1531 - read uid map: type g nsid 0 hostid 100000 range 1003
            lxc 20170323224319.395 INFO     lxc_confile - confile.c:config_idmap:1531 - read uid map: type g nsid 1004 hostid 101004 range 64532

**Comment by brauner** [comment]
Right, the cgroup related failures are a red herring. The actual issue is the ID mapping that cannot be set up and as I said before, in this case `userns_exec_1()` will fail and will not be able to remove the cgroups.

**Comment by brauner** [comment]
Idk if it's worth it, but we could try to be smart and have `userns_exec_1()` try to find the mapping for `uid 0` and only setup this mapping and then try to delete the corresponding cgroups. But that seems like a lof of special-casing.

**Comment by gaetanquentin** [comment]
hum ok. so what we must do when we have this issue for now? reboot?

**Comment by brauner** [comment]
So exactly what did you do to "mess up" the uid mapping?

**Comment by gaetanquentin** [comment]
lxc config set mycontainer  raw.idmap 'both 1003 1003'

**Comment by gaetanquentin** [comment]
1003 exists on host and container

**Comment by brauner** [comment]
Have you tried removing the `raw.idmap` entry and then (re)starting the container?

**Comment by gaetanquentin** [comment]
yes, don't work. i have to reboot for it to work again

**Comment by gaetanquentin** [comment]
was true for 2.11 .. i have tested it again with 2.12 and it works without reboot

**Comment by brauner** [comment]
So I can do

```
 lxc config set <c> raw.idmap 'both 1003 1003'
```
and reproduce the start failure but setting
```
lxc config unset <c> raw.idmap
lxc start <c>
```
and everything works.

**Comment by brauner** [comment]
Which is a solution to the start problem only of course.

**Comment by gaetanquentin** [comment]
an idea why '1003 1003' make it fails?

**Comment by brauner** [comment]
I suspect you need a separate entry in `/etc/sub{u,g}id` for that. One sec, let me test.

**Comment by brauner** [comment]
sudo as root do:

```
echo 'root:1003:1' | tee -a /etc/subuid /etc/subgid
```

and try again.

**Comment by gaetanquentin** [comment]
what is the "1"?

**Comment by brauner** [comment]
The rang of the mapping. (`man subuid`)

**Comment by gaetanquentin** [comment]
ok. it works. i have gaps with namespaces ...i will read some doc about it. thanks a lot

**Comment by brauner** [comment]
@gaetanquentin, sure np.

**Comment by gaetanquentin** [comment]
last question : if i want to set 2 idmap? (not a range): how to to it with raw.idmap?

**Comment by brauner** [comment]
What exactly do you mean by that?

**Comment by gaetanquentin** [comment]
'both 1003 1003' and 'both 1020 1020' for example

**Comment by gaetanquentin** [comment]
what is a "blob" in configuration?

**Comment by brauner** [comment]
A "blob" is simply a multi-line yaml entry:

```
raw.idmap: |
  both 1003 1003
  both 1020 1020
```

**Comment by brauner** [comment]
At least that's how I remember it.

**Comment by gaetanquentin** [comment]
so to set a blob with config set raw.idmap ?
 config set raw.idmap '| both 1003 1003   both 1020 1020' ?

**Comment by brauner** [comment]
`echo -e "both 1003 1003\nboth 1020 1020" | lxc config set alp1 raw.idmap -`
Not sure if that will work out of the box though.

**Comment by gaetanquentin** [comment]
it works! great. i think this should be added to the idmap documentation. thanks a lot

**Comment by brauner** [comment]
np
